react/AddToCartButton.tsx

```diff
@@ -157,18 +170,27 @@ const AddToCartButton: FC<Props> = ({
     }
   }
 
+  /*
+   * If text if an empty string it should render the default message
+   */
   const availableButtonContent = (
     <div className={`${handles.buttonDataContainer} flex justify-center`}>
-      <FormattedMessage id="store/add-to-cart.add-to-cart">
-        {message => <span className={handles.buttonText}>{message}</span>}
-      </FormattedMessage>
+      {!text ? (
+        <FormattedMessage id="store/add-to-cart.add-to-cart">
+          {message => <span className={handles.buttonText}>{message}</span>}
+        </FormattedMessage>
+      ) : (
+        <span className={handles.buttonText}>{text}</span>
+      )}
     </div>
   )
 
-  const unavailableButtonContent = (
+  const unavailableButtonContent = !unavailableText ? (
     <FormattedMessage id="store/add-to-cart.label-unavailable">
       {message => <span className={handles.buttonText}>{message}</span>}
     </FormattedMessage>
+  ) : (
+    <span className={handles.buttonText}>{unavailableText}</span>
```

Comment:
Just a nitpick: why inverted conditions here (`!text`, `!unavailableText`)? I think it would make it easier to read if we just used `text` and `unavailableText`